util/compression_test.cc

```diff
@@ -220,6 +220,197 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
+      }
+      SyncPoint::GetInstance()->EnableProcessing();
+    }
+    num_keys_++;
+    return true;
+  }
+
+ private:
+  int window_;
+  int num_keys_;
+  const BlockBuilder& data_block_builder_;
+
+  static size_t cpu_time_, predicted_cpu_time_, predicted_io_bytes_;
+  static size_t output_size_;
+};
+size_t CostAwareTestFlushBlockPolicy::cpu_time_ = 0,
+       CostAwareTestFlushBlockPolicy::predicted_cpu_time_ = 0,
+       CostAwareTestFlushBlockPolicy::predicted_io_bytes_ = 0,
+       CostAwareTestFlushBlockPolicy::output_size_ = 0;
+class CostAwareTestFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
+ public:
+  explicit CostAwareTestFlushBlockPolicyFactory(const int window)
+      : window_(window) {}
+
+  virtual const char* Name() const override {
+    return "CostAwareTestFlushBlockPolicyFactory";
+  }
+
+  virtual FlushBlockPolicy* NewFlushBlockPolicy(
+      const BlockBasedTableOptions& /*table_options*/,
+      const BlockBuilder& data_block_builder) const override {
+    (void)data_block_builder;
+    return new CostAwareTestFlushBlockPolicy(window_, data_block_builder);
+  }
+
+ private:
+  int window_;
+};
+
+class DBCompresssionCostPredictor : public DBTestBase {
+ public:
+  Options options;
+  DBCompresssionCostPredictor()
+      : DBTestBase("db_cpuio_skip", /*env_do_fsync=*/true),
+        options(CurrentOptions()) {
+    options.compression_manager = CreateCostAwareCompressionManager();
+    if (ZSTD_Supported()) {
+      options.compression = kZSTD;
+      options.bottommost_compression = kZSTD;
+    }
+    auto statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
+    options.statistics = statistics;
+    options.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
+    BlockBasedTableOptions bbto;
+    bbto.enable_index_compression = false;
+    bbto.flush_block_policy_factory.reset(
+        new CostAwareTestFlushBlockPolicyFactory(10));
+    options.table_factory.reset(NewBlockBasedTableFactory(bbto));
+    DestroyAndReopen(options);
+  }
+};
+
+TEST_F(DBCompresssionCostPredictor, CostAwareCompressorManager) {
+  // making sure that the compression is supported
+  if (ZSTD_Supported()) {
+    const int kValueSize = 20000;
+    auto index_ = 0;
+    Random rnd(231);
+    auto value = rnd.RandomBinaryString(kValueSize);
+    auto window_size = 10;
+    auto window_write = [&]() {
+      for (auto i = 0; i < window_size; ++i) {
+        auto status = Put(Key(index_), value);
+        EXPECT_EQ(status.ok(), true);
+        index_++;
+      }
+    };
+    // This denotes the first window
+    // Mocked to have specific cpu utilization and io cost
+    window_write();
+    // check the predictor is predicting the correct cpu and io cost
+    window_write();
+    // In this winodw we mock for another alogrithm and compression level to
+    // have specific cpu utilization and io cost
+    window_write();
+    // check the predictor is predicting the correct cpu and io cost
+    window_write();
```

Comment:
No need for the 3rd 4th writes after https://github.com/facebook/rocksdb/pull/13711/files#r2165217248.